File: divide-and-conquer/sort-an-array.java
class Solution {

    static int[] merge(int[] first, int[] second){
        int[] mix = new int[first.length + second.length];
        int i = 0;
        int j = 0;
        int x = 0;
        while(i<first.length && j<second.length){
            if(first[i] < second[j]){
                mix[x++] = first[i++];
            }else{
                mix[x++] = second[j++];
            }
        }

        while(i<first.length){
            mix[x++] = first[i++];
        }
        while(j<second.length){
            mix[x++] = second[j++];
        }
        return mix;
    }

    public int[] sortArray(int[] nums) {
        if(nums.length == 1){
            return nums;
        }

        int mid = nums.length/2;
        int[] left = sortArray(Arrays.copyOfRange(nums, 0, mid));
        int[] right = sortArray(Arrays.copyOfRange(nums,mid,nums.length));

        return merge(left,right);
    }
}

File: divide-and-conquer/sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        return self.divide(nums)
    
    def divide(self,arr):
        if len(arr) == 1:
            return arr

        mid = len(arr)//2
        left = self.divide(arr[:mid])
        right = self.divide(arr[mid:])
        return self.merge(left,right)
    
    def merge(self,left,right):
        newarr = []

        i = 0
        j = 0
        while i < len(left) and j < len(right):
            if left[i] < right[j]:
                newarr.append(left[i])
                i+=1
            else:
                newarr.append(right[j])
                j+=1
        while i<len(left):
            newarr.append(left[i])
            i += 1
        while j < len(right):
            newarr.append(right[j])
            j += 1
        return newarr